builder/googlecompute/driver_gce.go

```diff
@@ -92,7 +93,18 @@ func NewClientOptionGoogle(account *ServiceAccount, vaultOauth string, impersona
 		opts = append(opts, option.WithTokenSource(ts))
 
 	} else if impersonatesa != "" {
-		opts = append(opts, option.ImpersonateCredentials(impersonatesa))
+		log.Printf("[INFO] Using Google Cloud impersonation mechanism")
+		ts, err := impersonate.CredentialsTokenSource(context.Background(), impersonate.CredentialsConfig{
+			TargetPrincipal: impersonatesa,
+			Scopes: []string{
+				"https://www.googleapis.com/auth/cloud-platform",
+				"https://www.googleapis.com/auth/userinfo.email",
```

Comment:
Can you tell me more about these scopes, should these be parameters ? Are they the sort of minimal requirement ?

Reply:
This is clearly an unwanted error on my part. I'm gonna fix that immediatly as it should not be hard-coded this way.
Thanks for the warning  ;)

Reply:
> Can you tell me more about these scopes, should these be parameters ? Are they the sort of minimal requirement ?

@azr change is now done. Please tell me if you think additional work is needed. Thanks.